Oracle/scripts/split_files.sh

```diff
@@ -0,0 +1,98 @@
+## ----- Function to move files
+move_file() {
+    local file=$1
+    local FLNAME=$(grep -o -P "(?<=<${STERM}> )(.*?\..*?)(?= </${STERM}>)" $file | cut -d . -f 2)
+    local DBNAME=$(grep -o -P "(?<=<${STERM}> )(.*?\..*?)(?= </${STERM}>)" $file | cut -d . -f 1)
+    mkdir -p "${OUTPUT_PATH}/${DBNAME}/${SSUFF}"
+    mv $file "${OUTPUT_PATH}/${DBNAME}/${SSUFF}/${FLNAME}.sql"
+}
+export -f move_file
+
+## ---- Split array     
+declare -a split_terms=(
+[0]="sc-table"
+[1]="sc-view"
+[2]="sc-function"
+[3]="sc-procedure" 
+[4]="sc-package" 
+[5]="sc-synonym" 
+[6]="sc-type"
+[7]="sc-index"
+[8]="sc-trigger"
+[9]="sc-sequence"
+[10]="sc-dblink"
+[11]="sc-queue_table" 
+[12]="sc-olap_cube"
+[13]="sc-materialized_view" 
+[14]="sc-queue"
+[15]="sc-analytic_view" 
+[16]="sc-operator"
+)
+
+declare -a split_suffix=(
+[0]="Tables"
+[1]="Views"
+[2]="Functions"
+[3]="Procedures" 
+[4]="Packages" 
+[5]="Synonyms" 
+[6]="Types"
+[7]="Indexes"
+[8]="Triggers"
+[9]="Sequences"
+[10]="DBlink"
+[11]="QUEUE_TABLES" 
+[12]="OLAP_CUBES"
+[13]="MATERIALIZED_VIEWS" 
+[14]="QUEUES"
+[15]="ANALYTIC_VIEWS" 
+[16]="OPERATORS"
+)
+
+## ---- Loop through and split
+for ((i=0;i<=16;i++)); 
```

Comment:
Can we use something like split_suffix.length instead of a literal value?

Reply:
unfortunately there is no .length or similar for .sh files
there are some approaches but it will make the code a little messy.

Reply:
unfortunately there is no .length property for these arrays 
I tried an approach of getting the length into a variable and then use the variable in the loop but it doesn't work, idk if the array type changes something. I know this is not a very elegant solution but it is more fail-safe for .sh files.